code/modules/projectiles/guns/energy.dm

```diff
@@ -369,14 +370,16 @@
 		cell.loc = get_turf(src) //Drop it on the ground.
 	else
 		user.put_in_hands(cell)
-
-	playsound(loc, unload_sound, 25, 1, 5)
-	user?.visible_message(span_notice("[user] unloads [cell] from [src]."),
-	span_notice("You unload [cell] from [src]."), null, 4)
 	cell.update_icon()
-	set_cell(null)
-	update_icon(user)
-
+	if(user)
+		user.visible_message(span_notice("[user] unloads [cell] from [src]."), span_notice("You unload [cell] from [src]."), null, 4)
+		set_cell(null)
+		playsound(user, unload_sound, 25, 1, 5)
+		update_icon(user)
+		user.hud_used.update_ammo_hud(user, src)
+	else
+		set_cell(null)
+		update_icon()
```

Comment:
```suggestion
	if(user)
		user.visible_message(span_notice("[user] unloads [cell] from [src]."), span_notice("You unload [cell] from [src]."), null, 4)
		playsound(user, unload_sound, 25, 1, 5)
		user.hud_used.update_ammo_hud(user, src)
	set_cell(null)
	update_icon(USER)
```

Reply:
Aha well this won't work correctly. Why? Because the ammo counter will read the cell *before* it's set to null which is inaccurate.

Reply:
Can't because 

`user.visible_message(span_notice("[user] unloads [cell] from [src]."), span_notice("You unload [cell] from [src]."), null, 4)`